### PR TITLE
update JSON schema for subworkflows meta.yml

### PR DESCRIPTION
**Pull request by mirpedrol**
Update the JSON schema to validate subworkflows meta.yml. This adds the structure field to describe the structure of input and output channels.

```yaml
input:
  - ch_fasta:
      description: A channel containing the reference fasta file.
      structure:
          - meta:
              type: map
              description: meta map
          - fasta:
              type: file
              description: fasta file
              path: "*.fasta"
  - ch_fai:
      description: channel containing fasta.fai file
      structure: 
          - meta:
              type: map
              description: meta map
          - fai:
              type: file
              description: fasta.fai file
output:
  - versions:
      description: channel containing software versions file
      structure:
          - versions.yml
                type: file
                description: file containing versions of the software used
  - mutect2_vcf:
      description: Channel containing a list of compressed vcf files
      structure:
          - meta:
              type: map
              description: meta map
          - vcf:
              type: file
              description: vcf file 
```

### CODE REVIEW

File: subworkflows/yaml-schema.json [hunk 1]
```diff
@@ -51,32 +51,51 @@
...
+                            "structure": {
                                 "type": "array",
-                                "description": "List of allowed values for the input channel",
+                                "description": "Structure of the input channel",
                                 "items": {
-                                    "type": ["string", "number", "boolean", "array", "object"]
-                                },
-                                "uniqueItems": true
+                                    "type": "object",
+                                    "structureProperties": {
```
Comment (nvnieuwk) [inline subworkflows/yaml-schema.json:63]: I've never seen the `structureProperties` keyword, is this some feature that I don't know about or did you mean `patternProperties` here?

Comment (mirpedrol) [reply subworkflows/yaml-schema.json]: my bad, fixed now, thanks!

**Comment by mirpedrol** [comment]
@nf-core-bot fix linting

**Review by nvnieuwk — APPROVED** [review]
LGTM!

**Comment by mirpedrol** [comment]
I have decided to update some subworkflows to test the JSON file and have them as examples. 
I was hoping that it wouldn't take too much and I could update all of them but let's see where do we get 😉

**Review by mashehu — APPROVED** [review]
If you could add the `language-server` link to the json schema at the start of all files to so we start adding them, that would be great

**Comment by mirpedrol** [comment]
Merging this to a dedicated branch `subworkflows_meta_yml`, the idea is to work on this branch during the hackathon and then merge this to master.

**Pull request merged**